tools/autograd/derivatives.yaml

```diff
@@ -1544,8 +1544,13 @@
   result: auto_linear
 
 - name: sum.dim_IntList(Tensor self, int[1]? dim, bool keepdim=False, *, ScalarType? dtype=None) -> Tensor
-  self: sum_backward(grad, self.sizes(), dim, keepdim)
-  result: auto_linear
+  dispatch:
+    Default:
+      self: sum_backward(grad, self.sizes(), dim, keepdim)
```

Comment:
nit: Could you add the following test to TestAutogradMultipleDispatch please?
```python
    def test_sum_not_saving(self, device):
        def foo(x):
            # Don't modify the input inplace
            x = x.clone()
            res = x.sum()
            x.add_(1)
            return res

        inp = torch.rand(2, device=device, requires_grad=True)
        # sum's input is not saved for regular Tensors
        foo(inp).backward()

        # sum's input is saved for nested Tensors
        nt = torch.nested_tensor([torch.rand(2), torch.rand(2)], device=device).requires_grad_()
        with self.assertRaisesRegex(RuntimeError, "has been modified inplace"):
            foo(nt).backward()
```

(I can't run it locally, so there might be a bug there. But something in the same spirit.